spl-ids/src/token.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::arithmetic_side_effects)]
```

Comment:
Since we're going to the trouble of moving stuff around, it would be nice not to have to have this module-wide allow. Can we at least allow on a line-by-line basis, with comments about why?
(edit) It looks like it's only the one place

Reply:
removed it in favor of wrapping_add which should be the fastest and is safe because the bounds are already checked beforehand